CMakeLists.txt

```diff
@@ -32,32 +32,58 @@ endif()
 set(CMAKE_INCLUDE_CURRENT_DIR ON)
 set(CMAKE_INCLUDE_CURRENT_DIR_IN_INTERFACE ON)
 
-add_definitions(-DHAVE_ZLIB)
+include(FetchContent)
+include(GNUInstallDirs)
+set(INSTALL_CONFIGDIR ${CMAKE_INSTALL_LIBDIR}/cmake/${NIFTI_NIFTILIB_NAME})
+
 
 # add option to build shared libraries. The default is OFF to maintain the
 # current build behavior
+set( 1)
+  set( 1)
+
+option( NIFTI_INSTALL_NO_DEVELOPMENT "ITK build development flag." OFF)
+option( NIFTI_INSTALL_NO_LIBRARIES "ITK build development flag." OFF)
 option( BUILD_SHARED_LIBS "Toggle building shared libraries." OFF)
+if(BUILD_SHARED_LIBS)
+  set(NIFTI_LIBRARY_PROPERTIES
+    BUILD_SHARED_LIBS TRUE
+    POSITION_INDEPENDENT_CODE TRUE)
+  endif()
+
+# Generate a mapping from targets that need to be built to the sub-directories
+# in which they are located.
+macro(INSERT_INTO_DIRMAP _KEY _VALUE)
+  set("modules_${_KEY}" "${_VALUE}")
+endmacro()
+INSERT_INTO_DIRMAP(utils utils)
+INSERT_INTO_DIRMAP(znz znzlib)
+INSERT_INTO_DIRMAP(nifti2io nifti2)
+INSERT_INTO_DIRMAP(niftiio niftilib)
+INSERT_INTO_DIRMAP(cifti cifti)
+INSERT_INTO_DIRMAP(nifticdf nifticdf)
+
+# Fetch and populate the content of each sub-directory to provide access to
+# targets during cmake configure time
+foreach(module nifticdf niftiio nifti2io cifti)
+  FetchContent_Declare(
+  ${module}
+  SOURCE_DIR ${CMAKE_CURRENT_SOURCE_DIR}/${modules_${module}})
+  FetchContent_GetProperties(${module})
+if(NOT ${${module}_POPULATED})
+  FetchContent_Populate(${module})
+  add_subdirectory(${${module}_SOURCE_DIR} ${${module}_BINARY_DIR})
+endif()
+endforeach()
+
 
-# Michael Hanke 2007-04-25 Note:
-# 'Utilities' is merged into the include install path to be able to easily
-# install into any other path if necessary. For the same reason lib, bin and
-# include dir for standalone builds are only set if not yet defined.
 
-###  USE AS PART OF ITK
+##  USE AS PART OF ITK
 if("${ITK_VERSION_MAJOR}" GREATER 0.0)
 #   message(STATUS "---------------------ITK ZLIB  ${ITK_ZLIB_LIBRARIES}")
   add_definitions(-DITKZLIB)
   set(PACKAGE_PREFIX "ITK")
   set(NIFTI_LIBRARY_PROPERTIES ${ITK_LIBRARY_PROPERTIES})
-  set(NIFTI_INSTALL_BIN_DIR "${ITK_INSTALL_BIN_DIR_CM24}")
-  set(NIFTI_INSTALL_LIB_DIR "${ITK_INSTALL_LIB_DIR_CM24}")
```

Comment:
Why is this removed?

Reply:
Overall my attempt here was to use "include(GnuInstallDirs)"  to more concisely set installation/include/binary directories in each sub-directory so that they could be built individually or could be built as part of the full library. I'm happy to undo this, I just didn't want to verbosely copy with the non-existence of these directories for each sub-build. Perhaps the ability to build each sub-directory is not at all desirable though and so worth this substantial change in existing behaviour (plus I'm not sure how to implement similar behaviour for windows)

Reply:
GnuInstallDirs is very unix /linux centric.  Mac bundles and windows don't follow those same conventions.

For tools that I know are only needed on unix, it is OK.  I'd prefer not use GnuInstallDirs for nifti.